pkg/controller/statusmanager/machineconfig_watcher.go

```diff
@@ -0,0 +1,156 @@
+package statusmanager
+
+import (
+	"context"
+
+	"github.com/openshift/cluster-network-operator/pkg/platform"
+	"github.com/openshift/cluster-network-operator/pkg/util/k8s"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+type MachineConfigWatcher struct {
+	status *StatusManager
+	cache  cache.Cache
+}
+
+type MachineConfigPoolWatcher struct {
+	status *StatusManager
+	cache  cache.Cache
+}
+
+// AddMachineConfigWatcher wires up the MachineConfigWatcher to the controller-manager.
+func (s *StatusManager) AddMachineConfigWatcher(mgr manager.Manager) error {
+	if s.hyperShiftConfig.Enabled {
+		// MachineConfig is not supported in HyperShift cluster, so return without
+		// initializing watcher.
+		return nil
+	}
+
+	operatorCache := mgr.GetCache()
+	pw := &MachineConfigWatcher{
+		status: s,
+		cache:  operatorCache,
+	}
+	c, err := controller.New("machineconfig-watcher", mgr, controller.Options{Reconciler: pw})
+	if err != nil {
+		return err
+	}
+
+	return c.Watch(source.Kind[crclient.Object](operatorCache, &mcfgv1.MachineConfig{},
+		&handler.EnqueueRequestForObject{}, onMachineConfigPredicate()))
+}
+
+// Reconcile triggers a re-update of Status.
+func (p *MachineConfigWatcher) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	defer utilruntime.HandleCrash(p.status.SetDegradedOnPanicAndCrash)
+	mc := &mcfgv1.MachineConfig{}
+	err := p.cache.Get(ctx, request.NamespacedName, mc)
+	if err != nil && apierrors.IsNotFound(err) {
+		p.status.processDeletedMachineConfig(request.NamespacedName.Name)
+		return reconcile.Result{}, nil
+	}
+	if err != nil {
+		klog.Errorf("failed to retrieve machine config: %v", err)
+		return reconcile.Result{}, nil
+	}
+	p.status.processCreatedMachineConfig(*mc)
+	return reconcile.Result{}, nil
```

Comment:
This should just call SetFromMachineConfigPool. We don't really care about anything else.